src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -112,6 +112,22 @@ void gdb_value_extractort::analyze_symbols(
     CHECK_RETURN(value.pointee.empty() || (value.pointee == id));
 
     values.insert({value.address, symbol_expr});
+
+    const symbolt &symbol = ns.lookup(id);
+    const symbol_exprt actual_expr = symbol.symbol_expr();
```

Comment:
You can delay this until after the below `if`, because you can just use `symbol.type` instead of `actual_expr.type()`.

Reply:
Delayed.